rs/ic_os/config/src/types.rs

```diff
@@ -0,0 +1,84 @@
+use ic_types::malicious_behaviour::MaliciousBehaviour;
+use serde::{Deserialize, Serialize};
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::path::PathBuf;
+use url::Url;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct SetupOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub setupos_settings: SetupOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct HostOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct GuestOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct SetupOSSettings;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct HostOSSettings {
+    pub vm_memory: u32,
+    pub vm_cpu: String,
+    pub verbose: bool,
+}
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct NetworkSettings {
+    // Config files can specify ipv6_prefix and ipv6_gateway, or just an ipv6_address.
+    // ipv6_address takes precedence. Some tests provide only ipv6_address.
```

Comment:
I copied this comment from gary's networking code (code that we'll eventually delete once the revamp is done). The comment had a typo, so I wasn't certain if this was right. Here's the original comment:
```
// Config files can specify ipv6 prefix, address and prefix, or just address.
// ipv6_address takes precedence. Some tests provide only the address.
```